tests/common/dualtor/tor_failure_utils.py

```diff
@@ -135,11 +137,36 @@ def wait_for_device_reachable(duthost, timeout=300):
         if res.is_failed or ('msg' in res and 'Timeout' in res['msg']):
             raise Exception("DUT {} did not startup after reboot"
                             .format((duthost.hostname)))
-        logger.info("SSH started on {}".format((duthost.hostname)))
+        logger.info("SSH started on {}, wait for mux container to start".format((duthost.hostname)))
 
     return wait_for_device_reachable
 
 
+@pytest.fixture
+def wait_for_mux_container(localhost):
+    """
+    Returns a function that waits for mux container to be available on a device
+    """
+
+    def wait_for_mux_container(duthost, timeout=100, check_interval=1):
+        logger.info("Waiting for mux container to start on {}"
+                    .format((duthost.hostname)))
+        wait_time = 0
```

Comment:
can we use `wait_until`?
https://github.com/sonic-net/sonic-mgmt/blob/b447ec12731c88232e61a775f3b30fe217f0ef9f/tests/common/utilities.py#L90

Reply:
And maybe we could add to check if `mux` feature is enabled first before waiting for mux container?

Reply:
I will look into how I can integrate with wait_unitl. 
And this util file is used only for dualtor where mux container is always enabled. Do we still need to check for `mux` feature?

Reply:
I updated diff to take care of: 1) check mux feature before waiting for mux container check 2) use wait_until for both mux feature and container checks.

With mux feature check however I am seeing below errors when tests are run. It is because database container takes few seconds to start and initialize after we get SSH connection.

Since this util is specific to dualtor, shall I remove mux feature check from PR to avoid these errors?
Or do you have any other suggestions?

```
20:30:09 utilities.wait_until                     L0118 ERROR  | Exception caught while checking check_mux_feature:Traceback (most recent call last):
  File "/data/tests/common/utilities.py", line 112, in wait_until
    check_result = condition(*args, **kwargs)
  File "/data/tests/common/dualtor/tor_failure_utils.py", line 163, in check_mux_feature
    output = duthost.shell("show feature status mux")['stdout_lines']
  File "/data/tests/common/devices/multi_asic.py", line 108, in _run_on_asics
    return getattr(self.sonichost, self.multi_asic_attr)(*module_args, **complex_args)
  File "/data/tests/common/devices/base.py", line 89, in _run
    raise RunAnsibleModuleFail("run module {} failed".format(self.module_name), res)
RunAnsibleModuleFail: run module shell failed, Ansible Results =>
{
    "changed": true, 
    "cmd": "show feature status mux", 
    "delta": "0:00:00.780049", 
    "end": "2023-04-07 20:29:03.632989", 
    "failed": true, 
    "invocation": {
        "module_args": {
            "_raw_params": "show feature status mux", 
            "_uses_shell": true, 
            "argv": null, 
            "chdir": null, 
            "creates": null, 
            "executable": null, 
            "removes": null, 
            "stdin": null, 
            "stdin_add_newline": true, 
            "strip_empty_ends": true, 
            "warn": true
        }
    }, 
    "msg": "non-zero return code", 
    "rc": 1, 
    "start": "2023-04-07 20:29:02.852940", 
    "stderr": "Traceback (most recent call last):\n  File \"/usr/local/bin/show\", line 5, in <module>\n    from show.main import cli\n  File \"/usr/local/lib/python3.9/dist-packages/show/main.py\", line 207, in <module>\n    if is_gearbox_configured():\n  File \"/usr/local/lib/python3.9/dist-packages/show/main.py\", line 152, in is_gearbox_configured\n    app_db = SonicV2Connector()\n  File \"/usr/lib/python3/dist-packages/swsscommon/swsscommon.py\", line 1589, in __init__\n    for db_name in self.get_db_list():\n  File \"/usr/lib/python3/dist-packages/swsscommon/swsscommon.py\", line 1525, in get_db_list\n    return _swsscommon.SonicV2Connector_Native_get_db_list(self)\nRuntimeError: Sonic database config file doesn't exist at /var/run/redis/sonic-db/database_config.json", 
    "stderr_lines": [
        "Traceback (most recent call last):", 
        "  File \"/usr/local/bin/show\", line 5, in <module>", 
        "    from show.main import cli", 
        "  File \"/usr/local/lib/python3.9/dist-packages/show/main.py\", line 207, in <module>", 
        "    if is_gearbox_configured():", 
        "  File \"/usr/local/lib/python3.9/dist-packages/show/main.py\", line 152, in is_gearbox_configured", 
        "    app_db = SonicV2Connector()", 
        "  File \"/usr/lib/python3/dist-packages/swsscommon/swsscommon.py\", line 1589, in __init__", 
        "    for db_name in self.get_db_list():", 
        "  File \"/usr/lib/python3/dist-packages/swsscommon/swsscommon.py\", line 1525, in get_db_list", 
        "    return _swsscommon.SonicV2Connector_Native_get_db_list(self)", 
        "RuntimeError: Sonic database config file doesn't exist at /var/run/redis/sonic-db/database_config.json"
    ], 
    "stdout": "", 
    "stdout_lines": []
}
```